web_widget_numeric_step/static/src/numeric_step.esm.js

```diff
@@ -0,0 +1,99 @@
+import {FloatField} from "@web/views/fields/float/float_field";
+import {_t} from "@web/core/l10n/translation";
+import {registry} from "@web/core/registry";
+
+export class NumericStep extends FloatField {
+    setup() {
+        super.setup();
+        this.throttleTimer = null;
+        this.lastAssignedValue = this.props.record.data[this.props.name];
+        this.tempValue = this.lastAssignedValue;
+    }
+    willUpdateProps(nextProps) {
+        super.willUpdateProps(nextProps);
+        const newVal = nextProps.record.data[nextProps.name];
+        if (newVal !== this.lastAssignedValue) {
+            this.lastAssignedValue = newVal;
+            this.tempValue = newVal;
+        }
+    }
+    get value() {
+        return this.tempValue;
+    }
+    _onStepClick(ev) {
+        const mode = ev.target.dataset?.mode;
+        this._doStep(mode);
+    }
+    _onKeyDown(ev) {
+        if (ev.key === "ArrowUp") {
+            this._doStep("plus");
+        } else if (ev.key === "ArrowDown") {
+            this._doStep("minus");
+        }
```

Comment:
When the user enters a number using the keyboard and changes focus, the value is lost. Could you please check this?

Reply:
Thanks! solved